module/zfs/spa.c

```diff
@@ -3261,6 +3317,7 @@ spa_add_l2cache(spa_t *spa, nvlist_t *config)
 			    ZPOOL_CONFIG_VDEV_STATS, (uint64_t **)&vs, &vsc)
 			    == 0);
 			vdev_get_stats(vd, vs);
+			spa_update_l2cache_stats_ex(vd, l2cache[i]);
```

Comment:
Maybe, but I'm not sure.  If you look at the case of the regular stats, `spa_add_l2cache()` looks up the vdev_stats struct from the existing nvlist, and updates it with new data, rather than adding a new vdev_stats nvlist (which is what `vdev_config_generate_stats()` does).  I assume the code did it this way for a reason, so I did it the same way for the extended stats.  I will have to test and see if `vdev_config_generate_stats()` will work here instead.